apps/jupyter/extensions/notebook-runner/src/iframe-api.ts

```diff
@@ -522,6 +580,129 @@ export class EmbeddedPythonCallbacks {
     }
     return files;
   }
+
+  private async writeCrtInternalTaskOnEditMode(
+    importedFiles: CrtInternalTask,
+  ): Promise<void> {
+    if (this.mode == Mode.edit) {
+      await this.putFileContents(
+        EmbeddedPythonCallbacks.taskTemplateLocation,
+        importedFiles.taskTemplate,
+      );
+
+      await this.createFolder(EmbeddedPythonCallbacks.dataLocation, "data");
+      await this.writeFolderContents(
+        EmbeddedPythonCallbacks.dataLocation,
+        importedFiles.data,
+      );
+
+      await this.createFolder(
+        EmbeddedPythonCallbacks.gradingDataLocation,
+        "grading_data",
+      );
+      await this.writeFolderContents(
+        EmbeddedPythonCallbacks.gradingDataLocation,
+        importedFiles.gradingData,
+      );
+
+      await this.createFolder(EmbeddedPythonCallbacks.srcLocation, "src");
+      await this.writeFolderContents(
+        EmbeddedPythonCallbacks.srcLocation,
+        importedFiles.src,
+      );
+
+      await this.createFolder(
+        EmbeddedPythonCallbacks.gradingSrcLocation,
+        "grading_src",
+      );
+      await this.writeFolderContents(
+        EmbeddedPythonCallbacks.gradingSrcLocation,
+        importedFiles.gradingSrc,
+      );
+
+      this.documentManager.openOrReveal(
+        EmbeddedPythonCallbacks.taskTemplateLocation,
+      );
+    } else {
+      await this.createFolder("/student", "student");
+      await this.createFolder("/autograder", "autograder");
+
+      await this.putFileContents(
+        EmbeddedPythonCallbacks.studentTaskLocation,
+        importedFiles.studentTask,
+      );
+
+      await this.putFileContents(
+        EmbeddedPythonCallbacks.autograderLocation,
+        importedFiles.autograder,
+      );
+
+      await this.createFolder(EmbeddedPythonCallbacks.dataLocation, "data");
+      await this.writeFolderContents(
+        EmbeddedPythonCallbacks.dataLocation,
+        importedFiles.data,
+      );
+
+      await this.createFolder(EmbeddedPythonCallbacks.srcLocation, "src");
+      await this.writeFolderContents(
+        EmbeddedPythonCallbacks.srcLocation,
+        importedFiles.src,
+      );
+
+      this.documentManager.openOrReveal(
+        EmbeddedPythonCallbacks.studentTaskLocation,
+      );
+    }
+  }
+
+  private async writeExternalCustomTaskOnEditMode(
+    importedFiles: ExternalCustomTask,
+  ): Promise<void> {
+    if (this.mode == Mode.edit) {
```

Comment:
A guard clause would make this much easier to read.

Also it seems to me that this check may be better performed at a higher level. When any import message is received in solveMode, we can immediately abort.